frontend/src/model/Tile.ts

```diff
@@ -0,0 +1,44 @@
+import Locations from '../constants/locations';
+
+type Edges = {
+  bottom: Locations;
+  left: Locations;
+  right: Locations;
+  top: Locations;
+};
+
+class Player {
+  //TODO: import Player model
+}
+class Tile {
+  public edges: Edges;
+  public placedBy: Player;
+  public placementTurn: number;
+  constructor(edges: Edges, player: Player, placementTurn: number) {
+    this.edges = edges;
+    this.placedBy = player;
+    this.placementTurn = placementTurn;
+  }
+
+  rotateLeft() {
+    let prevEdges = { ...this.edges };
```

Comment:
We have this line in both rotate methods - how about replacing it with separate private method?